app/services/irs_attempts_api/redis_client.rb

```diff
@@ -9,24 +9,23 @@ class RedisClient
       end
     end
 
-    def write_event(jti:, jwe:)
+    def write_event(jti:, jwe:, timestamp:)
+      key = key(timestamp)
       redis_pool.with do |client|
-        client.setex(jti, IdentityConfig.store.irs_attempt_api_event_ttl_seconds, jwe)
+        client.hset(key, jti, jwe)
+        client.expire(key, IdentityConfig.store.irs_attempt_api_event_ttl_seconds)
       end
     end
 
-    def delete_events(event_ids)
+    def read_events(timestamp:)
+      key = key(timestamp)
       redis_pool.with do |client|
-        client.del(*event_ids)
+        client.hgetall(key)
       end
     end
 
-    def read_events(count = 1000)
-      redis_pool.with do |client|
-        keys = client.scan(0, count: count).last.first(count)
-        next {} if keys.empty?
-        client.mapped_mget(*keys)
-      end
+    def key(timestamp)
+      timestamp.change(min: 0, sec: 0).iso8601
```

Comment:
`iso8601` formats can differ based on timezone, WDYT about `.in_time_zone('UTC')` for consistency?

I know that the controller uses `Time.zone.parse` but a later closer to the source would help ensure consistency for unit tests, etc

```
[1] pry(main)> t = Time.zone.now
=> Wed, 17 Aug 2022 15:40:56.856020000 UTC +00:00
[2] pry(main)> t.iso8601
=> "2022-08-17T15:40:56Z"
[3] pry(main)> t.in_time_zone('America/Los_Angeles').iso8601
=> "2022-08-17T08:40:56-07:00"
```

```suggestion
      timestamp.in_time_zone('UTC').change(min: 0, sec: 0).iso8601
```

Reply:
Yes, good catch, thank you